packages/material/src/complex/table-array.control.tsx

```diff
@@ -41,117 +55,325 @@ export const tableArrayTester: RankedTester = rankWith(10, and(
-                              schema={resolvedSchema}
-                              uischema={createControlElement(prop)}
-                              path={childPath}
-                            />
-                          </TableCell>
-                        );
-                      })
-                      .value()
-                  }
-                </TableRow>
-              );
-            })
-          }
+            {isEmptyTable ? <EmptyTable/> : <TableWithContent {...tableProps}/>}
           </TableBody>
         </Table>
+        <Dialog
+          open={this.state.openConfirmDelete}
+          keepMounted
+          onClose={handleClose}
+          aria-labelledby='alert-dialog-confirmdelete-title'
+          aria-describedby='alert-dialog-confirmdelete-description'
+        >
+          <DialogTitle id='alert-dialog-confirmdelete-title'>
+            {'Confirm Deletion'}
+          </DialogTitle>
+          <DialogContent>
+            <DialogContentText id='alert-dialog-confirmdelete-description'>
+              Are you sure you want to delete the {this.state.selected.length} selected objects?
+            </DialogContentText>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={handleClose} color='primary'>
+              No
+            </Button>
+            <Button onClick={removeItems} color='primary'>
+              Yes
+            </Button>
+          </DialogActions>
+        </Dialog>
       </div>
     );
   }
 }
+export interface TableProps extends ControlProps {
+  childErrors: ErrorObject[];
+}
+// FIXME copied from validation.ts
+const errorAt = instancePath => (state): ErrorObject[] => {
+  const path = instancePath + '.';
 
+  return _.filter(state.errors, (error: ErrorObject) => error.dataPath.startsWith(path));
+};
+const mapStateToTableControlProps = (state, ownProps) => {
+  const {data, errors, classNames, label, visible, enabled, id, path, inputs, required} =
+    mapStateToControlProps(state, ownProps);
+
+  const childErrors = errorAt(path)(getValidation(state));
+
+  return {
+    data,
+    errors,
+    classNames,
+    label,
+    visible,
+    enabled,
+    id,
+    path,
+    inputs,
+    required,
+    childErrors
+  };
+};
 export default registerStartupRenderer(
   tableArrayTester,
-  connect(mapStateToControlProps)(TableArrayControl)
+  connect(mapStateToTableControlProps)(TableArrayControl)
 );
+const TableToolbar = ({errors, childErrors, label, path, uischema,
+  numSelected, handleOpenConfirmDelete, dispatch}) => {
+  const controlElement = uischema as ControlElement;
+  const labelClass = JsonForms.stylingRegistry.getAsClassName('array-table.label');
+  const buttonClass = JsonForms.stylingRegistry.getAsClassName('array-table.button');
+  const labelObject = Helpers.createLabelDescriptionFrom(controlElement);
+  const allErrors = [].concat(errors).concat(childErrors.map(e => e.message));
+
+  // TODO duplicate code
+  const addItem = () => {
```

Comment:
If we split up this into presentational and container component (as mentioned earlier) we might also provide some table-related common functionality that is used by both the vanilla and the material renderer in order to avoid code duplication.